src/util_test.c

```diff
@@ -81,6 +83,36 @@ void TestHashalgFromString() {
 }
 
 
+
+void TestCacheUtilReadShadowFile() { 
```

Comment:
Looks like the leaked memory is just from the test and wouldn't show up in real usage. You need to `g_free()` shadowpath and `g_bytes_unref()` gbytes_example_hash and actual_hash when they aren't needed anymore. Doing it all at the bottom is fine.